src/Hooks/useReactionState.js

```diff
@@ -53,9 +55,42 @@ export default function useReactionState(rxnTarget) {
         newUserRxns,
         increment
       );
+      // Update notifications, if this type notifies.
+      if (rxnType.notifies) {
+        const notification = getNotification(user, rxnTarget, rxnType);
+        if (newValue) {
+          SaveNotification(notification, rxnTarget.ownerId);
+        } else {
+          DeleteNotification(notification, rxnTarget.ownerId);
+        }
+      }
     },
     [user?.uid, userRxns, rxnTarget]
   );
 
   return [userRxns, updateUserRxns];
 }
+
+/**
+ * Builds a notification for `user` reacting to `rxnTarget` w/ type `rxnType`.
+ * @param {import("firebase/auth").User} user
+ * @param {import("../Util/ReactionUtil").RxnTarget} rxnTarget
+ * @param {import("../Util/ReactionUtil").RxnType} rxnType
+ * @returns {import("../Util/NotificationUtil").Notification} A notification
+ *  object that can be saved to storage.
+ */
+function getNotification(user, rxnTarget, rxnType) {
+  // TODO Extract basic notification template logic to shared lib.
+  return {
+    interactorId: user.uid, // Person sending the emoji
+    action: rxnType.type,
+    docId: rxnTarget.notifData.docId,
```

Comment:
The key name of "docId" here bothers me a little, it's really the "entityId" now (which was needed to fix the `deleteNotification` bug you caught). But for the sake of not breaking all existing notifications I'm fine with leaving it as is.

Reply:
Yeah, I agree about that.  The trouble is that the actual document ids for comments are the author's uid, which means these ids are not globally unique (since the author can comment on more than one item).  Otherwise, I'd suggest to just put the review document id in there.  We could put a full path in there instead, `watchlistData/${listOwnerId}${listId}` for watchlists and `watchlistData/${listOwnerId}${listId}/reviews/${reviewAuthorId}` for watchlist reviews.  This would make it kinda docId-esque, and these paths have to be globally unique since they are firebase paths.  

Alternatively, we could either:
1) do a migration later and start writing this field in parallel as both `docId` and `entityId` then eventually remove the `docId` after running a script to update any existing notifications without an `entityId` field,
2) Do (1) above, but with `docPath` as the new field to migrate to, and use paths instead of those generated entityId strings,
3) Do either (1) or (2), but quickly delete all notifications and start over, allowing us to make breaking changes to the notification schema, instead of a migration, or
4) leave it as it is and it'll just be a nice piece of tech debt to confuse interns some day

Reply:
My vote is for #4. 

It seems we have bigger fish to fry at the moment, and this is just a minor inconvenience/naming nit. Functionally, everything's working well, so let's leave this to the interns!